pkg/storage/utils/volumes.go

```diff
@@ -0,0 +1,121 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors.
+ *
+ */
+
+package utils
+
+import (
+	"context"
+	"fmt"
+
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "kubevirt.io/api/core/v1"
+	snapshotv1 "kubevirt.io/api/snapshot/v1beta1"
+	"kubevirt.io/client-go/kubecli"
+
+	backendstorage "kubevirt.io/kubevirt/pkg/storage/backend-storage"
+)
+
+const (
+	FLAG_EXCLUDE_NONE    = 0
+	FLAG_EXCLUDE_BACKEND = 1 << iota
+	// TODO: Add other flags as needed (KERNEL_BOOT?)
+)
+
+func createPVCVolume(pvcName string) *v1.Volume {
+	return &v1.Volume{
+		Name: pvcName,
+		VolumeSource: v1.VolumeSource{
+			PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+				PersistentVolumeClaimVolumeSource: k8sv1.PersistentVolumeClaimVolumeSource{
+					ClaimName: pvcName,
+					ReadOnly:  false,
+				},
+			},
+		},
+	}
+}
+
+func needsBackendPVC(vmiSpec v1.VirtualMachineInstanceSpec, excludeFlags int) bool {
+	return backendstorage.IsBackendStorageNeededForVMI(&vmiSpec) && excludeFlags&FLAG_EXCLUDE_BACKEND == 0
+}
+
+// GetVolumes returns all volumes of the passed object, empty if it's an unsupported object
+func GetVolumes(obj interface{}, excludeFlags int, client ...kubecli.KubevirtClient) ([]v1.Volume, error) {
```

Comment:
With the comments above, then this could become:
```golang
func GetVolumes(obj interface{}, client *kubecli.KubevirtClient, opts ...VolumeOption) ([]v1.Volume, error) {
```
and when you call it you can do <GetVolumes(vmi, nil)`  or `GetVolumes(vmi, nil, storagetypes.WithBackend, storagetypes.WithBootkernel)`

Reply:
ack, thanks!